sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LongLivedMessageSubscriber.java

```diff
@@ -0,0 +1,166 @@
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.reactivestreams.Subscription;
+import reactor.core.Disposable;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Mono;
+
+import java.util.Queue;
+import java.util.Timer;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class LongLivedMessageSubscriber extends BaseSubscriber<ServiceBusReceivedMessageContext> {
+
+    private final ClientLogger logger = new ClientLogger(SynchronousMessageSubscriber.class);
+    private Disposable timeoutOperation;
+    private final AtomicBoolean isDisposed = new AtomicBoolean(false);
+    private Queue<SynchronousReceiveWork> workQueue = new ConcurrentLinkedQueue<>();
+    private SynchronousReceiveWork currentWork = null;
+    private final AtomicInteger wip = new AtomicInteger();
+    private volatile Subscription subscription;
+    private long prefetch;
+
+
+    LongLivedMessageSubscriber(long prefetch) {
+        this.prefetch = prefetch;
+    }
+
+    /**
+     * On an initial subscription, will take the first work item, and request that amount of work for it.
+     *
+     * @param subscription Subscription for upstream.
+     */
+    @Override
+    protected void hookOnSubscribe(Subscription subscription) {
+        if (this.subscription == null) {
+            this.subscription = subscription;
```

Comment:
On subscribing you need to request something or it'll throw a debug message.

Reply:
I mistook this for the rule about request has to be > 0. However, the documentation suggests `hookOnSubscribe` is overridden when you want to alter the request.
https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/BaseSubscriber.java#L80-L82